src/raftstore/store/peer.rs

```diff
@@ -546,6 +587,9 @@ impl Peer {
             try!(self.handle_raft_commit_entries(&ready.committed_entries))
         };
 
+        // handle follower readonly query
+        self.handle_readonly_query(&ready.read_states);
```

Comment:
you can drain the read states here, so we don't need clone in handle_readonly_query.